### ISSUE TITLE
Overview of my confusion as guidance to README updates

### ISSUE THREAD

**Issue by consideRatio** [2i2c-org/farallon-image#18]
Thank you for your work on this and making it public for me to learn from!

Looking to learn about how AWS infra is deployed I inspected this repo and figured it would be helpful if I write down my understanding of the repos content and such, which may help the README be improved a bit.

I understood procedural parts quite well, but the overview of what deployments were managed in this repo and with what cloud infra has been challenging to understand for me. Here are things I've come to understand that required some detective work.

There are two hubploy deployments:
- farallon
- ohw - with repo2docker built image
  - Github Actions automate deployment of this image published in ECR, but I only see it use GCP credentials in sops key.

There is cloud infra from GCP/AWS/Azure:
- Hubploy use sops to decrypt/encrypt, which use a GCP KMS key independent of deployment. The GCP ServiceAccount to use sops is provided with a GitHub repo secret.
- Amazon ECR is used as an image registry for farallon and ohw deployments.
- Azure container registry is used for the Helm chart image with a custom kubespawner version
- Azure infra resides in terraform folder
- AWS EKS eksctl cluster configuration resides in aws folder
- There is a custom image/hub/Dockerfile with a custom hub pod image published to ACR, is this built manually only?
- There is no [GitHub secrets defined for this repo or org](https://github.com/2i2c-org/pangeo-hubs/settings/secrets/actions), but there is a reference to it in the deployment workflow of ohw: `${{ secrets.SOPS_GCP_DECRYPT_KEY }}`.

**Comment by choldgraf** [comment]
@consideRatio I think we just haven't yet put in the work to document these hubs, since we've been in "just get stuff up and running" mode for a few weeks. I think we should probably shift into "document and stabilize" mode for a bit before we hit 2021 to make sure that we have our bearings.

A meta-point here that @yuvipanda and I were talking about this week, is that we need to put time into documenting the hub infrastructure that 2i2c is deploying, make sure there is a single "source of truth", and make sure that it's clear how others could figure out what's going on.

We should do the same for _development priorities_ so that we're all on the same page about what we're working towards next. This will make it easier for us to coordinate efforts, help one another, and not duplicate effort!

I was imagining building out "2i2c documentation" and hosting it as a Sphinx site at `about.2i2c.org`...we can use this as a team handbook, similar to how https://about.gitlab.com/ hosts their company handbook there. What do people think about this?

**Comment by consideRatio** [comment]
@choldgraf I think that would be very relevant to have since we aim for the infrastructure to be practically reproducible. Perhaps we could nest the [technology overview](http://2i2c.org/zero-to-bootcamp/tech/index.html) there for example.

**Comment by yuvipanda** [comment]
Thanks for finding all these, @consideRatio. 

This repository is kinda a mess! It was forked from github.com/utoronto-2i2c/jupyterhub-deploy/ in a hurry and set up, so a lot of things don't make sense :) 



> Hubploy use sops to decrypt/encrypt, which use a GCP KMS key independent of deployment. The GCP ServiceAccount to use sops is provided with a GitHub repo secret.

This is also automatically set if you've appropriate rights in the `two-eye-two-see` GCP project, which is how I'm doing access control for now.

> Azure container registry is used for the Helm chart image with a custom kubespawner version

The terraform setup is leftover from the utoronto deploy, and isn't used. Same for chartpress.

> There is no [GitHub secrets defined for this repo or org](https://github.com/2i2c-org/pangeo-hubs/settings/secrets/actions), but there is a reference to it in the deployment workflow of ohw: `${{ secrets.SOPS_GCP_DECRYPT_KEY }}`.

Yeah, mostly because this is still manually deployed and not automated at all.

In conclusion, I'd suggest looking at [the Toronto deployment](https://github.com/utoronto-2i2c/jupyterhub-deploy/) (AKS) or [pilot hubs](https://github.com/2i2c-org/pilot-hubs/) (GCP) for inspiration, and not this repository :D I hope this repo will get there, but not yet.

**Comment by consideRatio** [comment]
@yuvipanda thanks for taking the time to help me understand this setup!
